src/__tests__/ParseQuery-test.js

```diff
@@ -1864,4 +1864,68 @@ describe('ParseQuery', () => {
     });
   });
 
+  it('full text search', () => {
+    const query = new ParseQuery('Item');
+    query.fullText('size', 'small');
+
+    expect(query.toJSON()).toEqual({
+      where: {
+        size: {
+          $text: {
+            $search: {
+              $term: "small"
+            }
+          }
+        }
+      }
+    });
+  });
+
+  it('full text search sort', () => {
+    const query = new ParseQuery('Item');
+    query.fullText('size', 'medium');
+    query.ascending('$score');
+    query.select('$score');
+
+    expect(query.toJSON()).toEqual({
+      where: {
+        size: {
+          $text: {
+            $search: {
+              $term: "medium",
+            }
+          }
+        }
+      },
+      keys : "$score",
+      order : "$score"
+    });
+  });
+
+  it('full text search key required', (done) => {
+    try {
+      const query = new ParseQuery('Item');
+      query.fullText();
+    } catch (e) {
```

Comment:
Rather than just try/catching I would use `expect(...).toThrow('my error...')`, that would be much more explicit in terms of that we are expecting and what we are expecting to receive as the error. You can see an example in the [parse-server tests](https://github.com/parse-community/parse-server/blob/8c2c76dd269f5aaf71439acb7a42a971a6560e49/spec/AdaptableController.spec.js#L27...L29).